nucypher/blockchain/eth/actors.py

```diff
@@ -518,8 +526,39 @@ def __init__(self, is_me: bool, *args, **kwargs) -> None:
         self.policy_agent = ContractAgency.get_agent(PolicyManagerAgent, registry=self.registry)
         self.staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=self.registry)
         self.economics = TokenEconomicsFactory.get_economics(registry=self.registry)
+
+        # Staking via contract
+        self.allocation_registry = allocation_registry or InMemoryAllocationRegistry()  # FIXME: This is a workaround for regular stakers
+        self.beneficiary_address = None
+        self.user_escrow_agent = None
+        self.check_if_staking_via_contract(self.checksum_address)
+
+        # Check stakes
         self.stakes = StakeList(registry=self.registry, checksum_address=self.checksum_address)
 
+    # TODO: This function not only "checks". Find a better name
+    def check_if_staking_via_contract(self, checksum_address: str) -> Optional[str]:
+        if not checksum_address:
```

Comment:
What if it will look at `self.checksum_address` if `checksum_address` is `None`?